### PR TITLE
:memo: livecoding / nested models step

### PR DESCRIPTION
**Pull request by delphinemillet**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
...
  3. Here we simply capitalize the name and the job is done 🙌
 
 
 ### Nested models
+> With mobx-state-tree, it's possible to define models that depends on another
+
+#### [#38](https://github.com/Dedetat/powerpuff-yourself/pull/38/files) - Nest models
+> Now it's easy to create a Powerpuff, they are spreading and it's becoming a mess ! Let's create a store to bring back some order
+ 1. We create a new model of **Store**, called `Store` (yeah, still clever 🤓)
+ 2. This model has a list of powerpuff as attribute, its type is an array of Powerpuff
```
Comment (delphinemillet) [inline README.md]: `powerpuffs`

File: README.md [hunk 2]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
  6. We can now use a sentence to instanciate our powerpuff, far much simpler ✨
 
 #### [#37](https://github.com/Dedetat/powerpuff-yourself/pull/37/files) - afterCreate
-> Say that we want to capitalize the name given to a Powerpuff (yes everybody can be mistaken, even better ones 😉)
+> Say we want to capitalize the name given to a Powerpuff (yes everybody can be mistaken, even better ones 😉)
```
Comment (fabienjuif) [inline README.md]: Say we want Powerpuffs' names to always be capitalized ?

File: README.md [hunk 3]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
...
  1. `afterCreate` is called after the model instanciation 💁‍♂️
  2. It should be defined as actions to access the `self` of the instance
  3. Here we simply capitalize the name and the job is done 🙌
 
 
 ### Nested models
+> With mobx-state-tree, it's possible to define models that depends on another
```
Comment (fabienjuif) [inline README.md]: depends on others ?

File: README.md [hunk 4]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
...
+
+#### [#38](https://github.com/Dedetat/powerpuff-yourself/pull/38/files) - Nest models
+> Now it's easy to create a Powerpuff, they are spreading and it's becoming a mess ! Let's create a store to bring back some order
+ 1. We create a new model of **Store**, called `Store` (yeah, still clever 🤓)
```
Comment (fabienjuif) [inline README.md]: ...We create a new model, wich will be our centralized **Store**, called `Store`...

File: README.md (hunk 1 above)
Comment (fabienjuif) [reply README.md]: its type refers to the model in english no ?

File: README.md [hunk 5]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
...
+ 2. This model has a list of powerpuff as attribute, its type is an array of Powerpuff
+ 3. You can notice we introduced a new type : `optional`. It's like `maybe` but you can define the default value. Here we want an empty list of powerpuff if a Store is not given a snapshot
```
Comment (fabienjuif) [inline README.md]: Here we want an empty array of powerpuffs, if the given snapshot doesn't have one.

File: README.md [hunk 6]
```diff
@@ -97,12 +97,24 @@ Welcome to our source speech repository about [mobx-state-tree](https://github.c
...
+ 4. The model has also a function to add a new Powerpuff into the list
+ 5. Instanciate a store without snapshot, you get a Store with an empty list of Powerpuff 🎉
+ 6. Instanciate some powerpuffs using the store instance and its list of Powerpuff is filled ✨
```
Comment (fabienjuif) [inline README.md]: ... the store action `addPowerpuff` ...

**Review by fabienjuif — CHANGES_REQUESTED** [review]
❤️ 
Some comments

**Review by fabienjuif — APPROVED** [review]

**Pull request merged**